Ira/my-app-video/src/components/filmPage/FilmPage.tsx

```diff
@@ -0,0 +1,176 @@
+import React, {useEffect} from 'react';
+import {RouteComponentProps, useHistory, useParams} from "react-router-dom";
+import {Header} from "../layout/header";
+import {FullInfoFilmCard} from "../fullInfoFilmCard";
+import {Footer} from "../layout/footer";
+import {FilmList} from "../filmList";
+import {Button} from "../shared/button/Button";
+import Pagination from "../pagination";
+import {initMoviePage, resetMovies, fetchMovies} from "../../toolkitRedux/toolkitSlice";
+import {queryString} from "../../constants/queryString";
+import {SortBy} from "../sortingSection/SortingSection";
+import {SearchBy} from "../shared/searchFilmForm/SearchFilmForm";
+import loader from '../../accets/Dual Ball-1s-200px.svg';
+import {useSelector, useDispatch} from "react-redux";
+import {RootState} from "../../toolkitRedux";
+import './styles.scss';
+
+type FilmPageProps = RouteComponentProps<{ id: string }>;
+
+export const FilmPage = ({location}: FilmPageProps) => {
+  const dispatch = useDispatch();
+  const {id} = useParams<{ id: string }>();
+  const history = useHistory();
+  const loading = useSelector(((state: RootState) => state.store.loading));
+  const currentFilm = useSelector((state: RootState) => state.store.currentMovie);
+  const movies = useSelector((state: RootState) => state.store.movies);
+
+  useEffect(() => {
+    initPage();
+
+    return () => {
+      resetMovies();
+    }
+  }, []);
+
+  useEffect(() => {
+    initPage();
+  }, [location]);
+
+  const getParams = (): { page: number } => {
+    return queryString.parse(location.search) as
+      { page: number };
+  }
+  const params = getParams();
+
+  const initPage = (): void => {
+    dispatch(initMoviePage({page: params.page, searchBy: SearchBy.genre, id: Number(id)}));
+  }
+
+  const onPage = (btnId: number) => {
+    let genre: string = '';
+
+    if (currentFilm) {
+      genre = currentFilm.genres[0].toLocaleLowerCase();
+    }
+
+    const newParams: string = queryString.stringify({page: btnId});
+    const payLoadParams: { search: string, searchBy: string, sortBy: string, page: number } = {
+      search: genre, searchBy: SearchBy.genre, sortBy: SortBy.release, page: btnId || params.page
+    };
+
+    history.push(`/film/${Number(id)}?${newParams}`);
+    dispatch(fetchMovies(payLoadParams));
```

Comment:
useEffect that subscribed on location will be called when you do history.push. You have an issue in logic